lib/internal/child_process.js

```diff
@@ -78,21 +78,26 @@ const handleConversion = {
         if (firstTime) socket.server._setupSlave(socketList);
 
         // Act like socket is detached
-        socket.server._connections--;
+        if (!options.keepOpen)
+          socket.server._connections--;
+
       }
 
+      var handle = socket._handle;
+
       // remove handle from socket object, it will be closed when the socket
       // will be sent
-      var handle = socket._handle;
-      handle.onread = function() {};
-      socket._handle = null;
+      if (!options.keepOpen) {
+        handle.onread = function() {};
+        socket._handle = null;
+      }
 
       return handle;
     },
 
-    postSend: function(handle) {
+    postSend: function(handle, options) {
       // Close the Socket handle after sending it
```

Comment:
I don't really like the idea of returning from the function early if `options.keepOpen` is `true`. That could require more refactoring if we ever decided to add more functionality here. It also makes the code less readable IMO. Also, 100% of legacy code and probably the majority of new code would have `keepOpen = false`, meaning that it would have to run two separate `if` statements instead of only checking `keepOpen` if `handle` is truthy.